teps/0152-tekton-conformance-verification.md

```diff
@@ -0,0 +1,219 @@
+
+![conformance verification](images/0152-conformance-verification.png)
+
+More specifically for the verification process:
+- The test suite provides Tekton Primitives YAMLs as inputs for the vendor service to process. The YAMLs are vanilla Tekton-native PipelineRuns and TaskRuns that are vendor-agnostic. Each field/feature will be isolated in separate tests as much as possible to simplify failure attribution.
+
+- The black box testing of the vendor focuses solely on the outputs of the service, disregarding its internal implementation details. The vendor service takes ownership of monitoring CRD reconciliation and delivering compatible Tekton YAML outputs.
+
+- For verifying results, upon receiving [Tekton client](https://github.com/tektoncd/pipeline/blob/main/test/clients.go#L56-L71) readable YAML outputs,  the test suite will leverage the parse pkg for validating whether all the required conformant API fields are populated and the behaviors are honored.
+
+#### Vendor integration
+
+Tekton vendors will have the responsibility of integrating the conformance verification test suite with their respective services. To deliver conformance validation results, vendors must implement the `ProcessAndSendToTekton` function. On a higher level, it is a wrapper of all the vendor service detail, intaking and outputing Tekton YAML for Tekton conformance fields to get verified.
+
+This function transforms vanilla Tekton Pipeline YAMLs into Tekton-parsable YAMLs for results verification. For more details:
+
+```go
+// ProcessAndSendToTekton accepts vanilla Tekton PipelineRun and TaskRun YAMLs, waits for the object to complete and outputs the completed PipelineRun and TaskRun along with the status.
+
+// The function parameters are the inputYAML and its corresponding Primitive type (either {PipelineRun} or {TaskRun}). Additionally, the function allows for the inclusion of custom inputs as variadic parameters.
+
+// The function will return the completed output YAML string and any associated errors.
+func ProcessAndSendToTekton(inputYAML, type string, customInputs …interface{}) (string, error)
+```
+
+#### **Alternatives Considered**
+As an alternative, Tekton vendors could expose endpoints for Tekton to call, allowing for  consumption and output of results according to HTTP API conformance (out-of-scope for the existing Tekton Conformance policy). In this scenario, Tekton Pipeline could trigger the testing process autonomously.
+
+##### Cons:
+- This would require HTTP API conformance which is not currently in-scope with [v1 Tekton conformance API spec](https://github.com/tektoncd/pipeline/blob/main/docs/api-spec.md).
+- This increases complexity and maintenance costs for the Tekton OSS project. Vendors seeking conformance should be responsible for managing these costs.
+
+### Vendor Requirements
+- Vendor service should handle conformant vanilla PipelineRun and TaskRun YAMLs successfully, ensuring all the REQUIRED fields are accepted.
```

Comment:
are we using only pipelinerun/taskruns? what about just pipelines and tasks?

Reply:
It makes sense to have all Task, Pipeline, TaskRun and PipelineRun. Updated. Thanks Dibyo.